### PR TITLE
Enable gRPC reflection

### PR DESCRIPTION
**Pull request by codingllama**
It's now possible to use reflect-enabled clients such as grpc_cli.

Examples:

```
$ grpc_cli call localhost:1234 ListClusters ''
$ grpcurl -plaintext localhost:1234 \
  teleport.terminal.v1.TerminalService/ListClusters
```

### CODE REVIEW

**Comment by zmb3** [comment]
Are there any security concerns with enabling server reflection?

File: lib/terminal/server.go [hunk 1]
```diff
@@ -64,6 +65,9 @@ type ServerOpts struct {
 	// RuntimeOpts that is sent before the server is started.
 	ConfigOutput io.Writer `json:"-"`
 
+	// DisableReflect disables gRPC reflection.
+	DisableReflect bool `json:"disable_reflect"`
```
Comment (zmb3) [inline lib/terminal/server.go:69]: Any particular reason to make it opt-out vs opt-in?

Comment (codingllama) [reply lib/terminal/server.go:69]: Opt-out is for the paranoid, opt-in so I need less flags to play with my local server. :)

**Comment by codingllama** [comment]
> Are there any security concerns with enabling server reflection?

I don't think so [1]. The API we are exposing is the same and the protos are not secret (given that we are open source and all), we just made it a bit easier to interact with. (It's not like hiding the protos would protect the API anyway.)

mTLS is still possible, and that is what one should shoot for if they don't want 3rd parties interacting with the API. Using domain sockets is also possible, which I assume narrows down access as well.

[1] This is in no way a guarantee of anything :)

**Review by zmb3 — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks for the quick review.

**Pull request merged**